packages/bridge/module.cjs

```diff
@@ -26,6 +26,13 @@ module.exports.defineNuxtConfig = (config = {}) => {
     // Ensure other modules register their hooks before
     config.buildModules.push('@nuxt/bridge')
   }
+
+  // Turn on vue-router if pages is enabled. Backport of nuxt 3 behavior
+  if (config.pages && typeof config.build?.createRoutes !== 'function') {
```

Comment:
```suggestion
  if (!config.pages && typeof config.build?.createRoutes !== 'function') {
```

Reply:
My bad, I needed to link a code I found before creating this PR
https://github.com/nuxt/nuxt/blob/2.x/packages/builder/src/builder.js#L178-L197

If we have config.build.createRoutes then we have pages support even without /pages directory, just like `pages: true` option works in Nuxt 3. But if Nuxt 2 user sets `pages: true`, we need to "mock" `config.build.createRoutes` function to bypass this check and allow to render pages other than default.

The code I propose works in 2.16 edge. I want to check tomorrow if there any changes with stable 2.16, but IMO `config.pages` shouldn't be `false` to enable this feture

Reply:
I created small playground, which prooves my opinion:
https://github.com/AndreyYolkin/nuxt-bridge-page-playground
https://codesandbox.io/s/github/AndreyYolkin/nuxt-bridge-page-playground

Just set keys in config and see the result.

Have a nice conf!

Reply:
Seems like we are looking to the problem from differrent sides. I want to extend pages via @nuxt/kit's extendPages method without keeping `pages` folder with .gitkeep file, like how it works in [Nuxt 3](https://nuxt.com/docs/api/configuration/nuxt-config#pages-1) (That's why I proposed `pages: true`). You want to disable vue-router when setting pages to false (that's why you proposed `pages: false`)

Without pages folder and config.build.createRoutes you [will get](https://github.com/nuxt/nuxt/blob/2.x/packages/builder/src/builder.js#L178-L197) that default page (Layout is coral) and function returning empty array of routes just overrides that behaviour.
![image](https://user-images.githubusercontent.com/11289484/235891627-b404936e-63c8-43a2-a12c-d40051e9108f.png)

If we wants to disable vue-router with `pages: false` we also need to give people ability to override that default page by custom (remember we discussed backporting `app.vue`, IMO it's inevitable, if some wants to use Nuxt 2b without pages)

Reply:
ahhh, I see what you mean.